pkg/api/types.go

```diff
@@ -186,6 +187,17 @@ type VMDiagnostics struct {
 // OrchestratorType defines orchestrators supported by ACS
 type OrchestratorType string
 
+// OrchestratorTypeEqualer compares OrchestratorType
+type OrchestratorTypeEqualer interface {
```

Comment:
Discussed offline. Adding a helper function makes the code concise. My suggestion is to convert the name of a generic StringComparer so it can be reused.